In [7]:
import seaborn as sns
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt

In [8]:
data = pd.read_csv('./credit_scoring_eng.csv')
df = pd.DataFrame(data)
df2 = pd.DataFrame(data)

In [9]:
df.head()

children  days_employed  dob_years            education  education_id  \
0         1   -8437.673028         42       masters degree             0   
1         1   -4024.803754         36  secondary education             1   
2         0   -5623.422610         33  Secondary Education             1   
3         3   -4124.747207         32  secondary education             1   
4         0  340266.072047         53  secondary education             1   

       family_status  family_status_id gender income_type  debt  \
0            married                 0      F    employee     0   
1            married                 0      F    employee     0   
2            married                 0      M    employee     0   
3            married                 0      M    employee     0   
4  civil partnership                 1      F     retiree     0   

    total_income                  purpose  
0  253875.639453    purchase of the house  
1  112080.014102             car purchase  
2  145885.952297    purchase of the house  
3  267628.550329  supplementary education  
4  158616.077870        to have a wedding

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


### children
##### Número de filho

### days_employed
##### Dias empregados, sendo negativo os dias desempregados.

### dob_years
##### Idade da pessoa

### education
##### Nivel de escolaridade da pessoa

### education_id
##### Id do nível de escolaridade

### family_status
##### Status civil da pessoa

### family_status_id
##### Id do status civil

### gender
##### Genero da pessoa

### income_type
##### Tipo e renda, se é trabalhador ou aposentado

### debt
##### se tem divida ou não, sendo 0 não e 1 sim

### total income
##### total da renda

### purpose
##### proposito do pq querer mais credito.
ito do pq querer mais credito






In [11]:
df.isna()

children  days_employed  dob_years  education  education_id  \
0         False          False      False      False         False   
1         False          False      False      False         False   
2         False          False      False      False         False   
3         False          False      False      False         False   
4         False          False      False      False         False   
...         ...            ...        ...        ...           ...   
21520     False          False      False      False         False   
21521     False          False      False      False         False   
21522     False          False      False      False         False   
21523     False          False      False      False         False   
21524     False          False      False      False         False   

       family_status  family_status_id  gender  income_type   debt  \
0              False             False   False        False  False   
1              False             False   False        False  False   
2              False             False   False        False  False   
3              False             False   False        False  False   
4              False             False   False        False  False   
...              ...               ...     ...          ...    ...   
21520          False             False   False        False  False   
21521          False             False   False        False  False   
21522          False             False   False        False  False   
21523          False             False   False        False  False   
21524          False             False   False        False  False   

       total_income  purpose  
0             False    False  
1             False    False  
2             False    False  
3             False    False  
4             False    False  
...             ...      ...  
21520         False    False  
21521         False    False  
21522         False    False  
21523         False    False  
21524         False    False  

[21525 rows x 12 columns]

In [12]:
df.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

In [13]:
df.isna().sum() / len(df)

children            0.000000
days_employed       0.100999
dob_years           0.000000
education           0.000000
education_id        0.000000
family_status       0.000000
family_status_id    0.000000
gender              0.000000
income_type         0.000000
debt                0.000000
total_income        0.100999
purpose             0.000000
dtype: float64

####  Quais colunas tem valores ausentes? Qual percentual 
de valores ausentes para cada coluna
days_employed (percentual 0.100999) e total_income (percentual: 0.100999)?